src/lib/loaders/loaders_with_authentication/gravity.ts

```diff
@@ -185,6 +185,7 @@ export default (accessToken, userID, opts) => {
       {},
       { headers: true }
     ),
+    salesLoaderWithHeaders: gravityLoader("sales", {}, { headers: true }),
```

Comment:
So the reason for the `WithHeaders` name in the loader is that there is another, non-authenticated loader without headers?

Reply:
Yes, the name should match so (I think) the resolver picks up the right one and you don't have to worry about doing it by myself.

Reply:
Yup, this is the preferred way to do it, if you have a need for using both authenticated (passing along a user's token, and not caching within MP), and unauthenticated (not passing along the user's token and subject to caching). Only something that's user-specific (so requiring the authenticated one), but could also be used without a user, would make sense to set up in this fashion.

You name them the same, and if the user makes a request to MP while logged in, the loader with authentication gets automatically selected. However, the context for the request also contains a map of `authenticatedLoaders` and `unauthenticatedLoaders`, so one could choose to use one or the other based on the request.

I'm possibly missing context here, but it seems like the `registered` argument is a possible argument to the API which is user-specific, otherwise this can be cached?

If so, doing it this way will wind up skipping caching _all_ calls that use the `salesLoaderWithHeaders` loader whenever a user is logged in. That might be fine, but we have a pattern for choosing to use the authenticated one (which skips caching), only when needed (due to a particular query param only making sense for a logged-in user, for instance).

An example of that is in https://github.com/artsy/metaphysics/blob/3af6a8971b010d574431e89bd696ea9fede4ea70/src/schema/v2/filterArtworksConnection.ts#L468-L478

There, we'd like to cache the results of the filter artworks call (that unauthenticated loader in particular is cached for longer than others), and only when there are particular arguments, do we want to skip the MP cache and use the authenticated loader.

Reply:
That is really interesting, thanks for the info @mzikherman! I've just updated how it picks up the loader as well as a test that covers that case.

In terms of the error of `"You must be logged in to request these params."`, I was just about 50:50 on this. It would be useful to get a warning if I'm using the argument with a necessary auth token, but at the same time it might be a bit too disruptive to force-return an error when the frond-end can continue to render the page.